### DISCUSSION TITLE
Pagination: increment Page + 1

### DISCUSSION THREAD

**Discussion by Mkallday** [thunderclient/thunder-client-support#1632] [Q&A]
Request Url: https://ssapi.shipstation.com/orders?orderStatus=awaiting_shipment&page={{1|add("page")}}&pageSize=20

# TEST --> SCRIPTING

```typescript
let incrementCount = parseInt(tc.getVar('incrementCount') || "0");
let code = tc.response.status;
if (incrementCount <= 3 && code !== 200) {
    incrementCount = incrementCount + 1;
    tc.setVar('incrementCount', incrementCount);
    console.log("Retrying request", incrementCount);

    await tc.delay(incrementCount * 1000); // exponential delay of 1 sec
    await tc.retryRequest();  // retry request
} else {
    tc.setVar('incrementCount', 0);
    console.log("Reset incrementCount = 0");
}

```
# RESPONSE:

```
{
  "Message": "The request is invalid.",
  "ModelState": {
    "page": [
      "The value '{{1|add(\"page\")}}' is not valid for Int32."
    ]
  }
}

```

**Comment by Mkallday** [comment]
@rangav I'm using the paid version. Should I email you the license number again?

**Reply by rangav** [reply]
sorry for the delay, will verify and get back to you

**Reply by rangav** [reply]
It is better we do a call and discuss the issue.

are you available now?

confirm your availability using below form
https://www.thunderclient.com/contact

**Comment by rangav** [comment]
use this url

https://ssapi.shipstation.com/orders?orderStatus=awaiting_shipment&page={{page}}&pageSize=20
